node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -333,11 +306,28 @@ impl PartialOrd for CandidateComparator {
 
 impl Ord for CandidateComparator {
 	fn cmp(&self, other: &Self) -> Ordering {
-		match self.relay_parent_block_number.cmp(&other.relay_parent_block_number) {
-			Ordering::Equal => (),
-			o => return o,
+		return match (self.relay_parent_block_number, other.relay_parent_block_number) {
+			(None, None) => {
+				// No relay parents for both -> compare hashes
+				self.candidate_hash.cmp(&other.candidate_hash)
+			},
+			(Some(_), Some(_)) => {
```

Comment:
Here's a refactor that would look more natural to me:

```
(Some(self_parent_number), Some(other_parent_number)) => {
    match self_parent_number.cmp(other_parent_number) => {
        // These lines same
    }
}
```

I can see why you might want to stick with the nice look of `(Some(_), Some(_))` though

Reply:
Your suggestion looks better, thanks!